app/src/main/java/com/SecUpwN/AIMSICD/adapters/AIMSICDDbAdapter.java

```diff
@@ -138,10 +142,12 @@
     private String[] mTables;
     private SQLiteDatabase mDb;
     private Context mContext;
+    private SharedPreferences mPreferences;
 
     public AIMSICDDbAdapter(Context context) {
         super(context, DB_NAME, null, 1);
         mContext = context;
+        mPreferences = PreferenceManager.getDefaultSharedPreferences(mContext);
```

Comment:
It doesn't matter how often the constructor is called.  

`private SharedPreferences mPreferences;` it's not static final so the object will be reconstructed every time a compiler/method is called. Field can be limited to method-body or classifier changer to static final to limit resource usage.